### ISSUE TITLE
kepextract seems to be slow on short cadence data

### ISSUE THREAD

**Issue by mirca** [KeplerGO/PyKE#22]

**Comment by mirca** [comment]
The slowest part of ``kepextract`` is this ``for`` loop

```
    for i in tqdm(range(ntime)):
        modf = np.zeros(shape=(naper))
        k = -1
        guess = [mom_centr1[i], mom_centr2[i], np.nanmax(flux[i:]), 1.0, 1.0, 0.0, 0.0]
        for j in range(len(aperb)):
            if (aperb[j] == 3):
                k += 1
                modf[k] = flux[i,j]
                args = (modx, mody, modf)
        try:
            ans = leastsq(kepfunc.PRFgauss2d, guess, args=args, xtol=1.0e-8,
                          ftol=1.0e-4, full_output=True)
            s_sq = (ans[2]['fvec'] ** 2).sum() / (ntime - len(guess))
            psf_centr1[i] = ans[0][0]
            psf_centr2[i] = ans[0][1]
        except:
            pass
        try:
            psf_centr1_err[i] = sqrt(diag(ans[1] * s_sq))[0]
        except:
            psf_centr1_err[i] = np.nan
        try:
            psf_centr2_err[i] = sqrt(diag(ans[1] * s_sq))[1]
        except:
            psf_centr2_err[i] = np.nan
```

**Comment by mrtommyb** [comment]
It's because kepextract was modified to do the PSF photometry. You should probably make this optional, or spin it off into a different module.

**Comment by mirca** [comment]
@mrtommyb oh yeah, thanks Tom!

I guess this moment generation data part can also be suppressed, what do you think?
```
    # construct new table moment data
    mom_centr1 = np.zeros(shape=(ntime))
    mom_centr2 = np.zeros(shape=(ntime))
    mom_centr1_err = np.zeros(shape=(ntime))
    mom_centr2_err = np.zeros(shape=(ntime))
    for i in tqdm(range(ntime)):
        xf = np.zeros(shape=(naper))
        yf = np.zeros(shape=(naper))
        f = np.zeros(shape=(naper))
        xfe = np.zeros(shape=(naper))
        yfe = np.zeros(shape=(naper))
        fe = np.zeros(shape=(naper))
        k = -1
        for j in range(len(aperb)):
            if aperb[j] == 3:
                k += 1
                xf[k] = aperx[j] * flux[i, j]
                xfe[k] = aperx[j] * flux_err[i, j]
                yf[k] = apery[j] * flux[i, j]
                yfe[k] = apery[j] * flux_err[i, j]
                f[k] = flux[i, j]
                fe[k] = flux_err[i, j]
        xfsum = np.sum(xf)
        yfsum = np.sum(yf)
        fsum = np.sum(f)
        xfsume = math.sqrt(np.sum(xfe * xfe) / naper)
        yfsume = math.sqrt(np.sum(yfe * yfe) / naper)
        fsume = math.sqrt(np.sum(fe * fe) / naper)
        mom_centr1[i] = xfsum / fsum
        mom_centr2[i] = yfsum / fsum
        mom_centr1_err[i] = math.sqrt((xfsume / xfsum) ** 2 + ((fsume / fsum) ** 2))
        mom_centr2_err[i] = math.sqrt((yfsume / yfsum) ** 2 + ((fsume / fsum) ** 2))
    mom_centr1_err = mom_centr1_err * mom_centr1
    mom_centr2_err = mom_centr2_err * mom_centr2
```

**Comment by mrtommyb** [comment]
I think making it optional is sensible, and then in the description warn that it is slow for short cadence data.

**Comment by mirca** [comment]
@mrtommyb awesome!

**Comment by mirca** [comment]
closed by #44

**Comment by mirca** [comment]
thanks @jjhermes
